language/move-ir/types/src/ast.rs

```diff
@@ -149,13 +152,16 @@ pub type TypeVar = Spanned<TypeVar_>;
 // TODO: This enum is completely equivalent to vm::file_format::Kind.
 //       Should we just use vm::file_format::Kind or replace both with a common one?
 /// The kind of a type. Analogous to `vm::file_format::Kind`.
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[derive(Debug, Display, PartialEq, Eq, Clone)]
 pub enum Kind {
     /// Represents the super set of all types.
+    #[display(fmt = "all")]
```

Comment:
I removed this a while back because I felt it hurt the readability of the AST. Thoughts on reverting?

Reply:
Do you mean the declarative annotations? Personally I strongly prefer them to a separate impl because it's all in one place.

Reply:
What does all in one place mean here?

Reply:
The declaration and the Display annotation live together. I think of the Display impl as providing a lot of valuable information to understand the code — I looked at them several times while figuring out how the AST IR worked. But I might be weird 🤷

Reply:
My main worry is some of them get quite verbose. If they were all one line, I would consider using them to replace the syntax/grammar comments

But as it stands, I'd prefer to keep them separate

Reply:
Well, I definitely prefer to keep them together — I think most of them are one line.

Reply:
FWIW it wouldn't be a good idea to replace the docs either — cargo doc is a super valuable way to look at these giant AST enums.

Reply:
> Well, I definitely prefer to keep them together 

I just don't see much benefit from using the derivation annotation. I made the switch to remove them as I wanted to fully separate the AST from any impls, as I found it made it easier to grok the AST. I'd like to keep that separation in place

Reply:
I understand that, but my view is completely the opposite. I find it unnecessarily complex to have the AST be separate from any impls, and I found it to be much harder to grok the AST with the separate impls. I strongly prefer making this change.

Out differences might be irreconcilable. How do you wish to proceed?

Reply:
All of the ASTs in the source language follow this same pattern. So for now, differences aside, I think the best thing to do is to be consistent

Reply:
I'm happy to make the same change to the source language ASTs as well, as much as it makes sense.

Reply:
I don't really value the change for the source language ASTs. And I'm really the only person using that code at the moment, so I'd like to keep it the way that let's me move as fast as possible

Reply:
Side note: I'm trying my best to understand your point of view, but I'm very surprised at the pushback to this. To me this is just plainly better. It is more declarative, it minimizes repeating oneself, and it really shines for larger enums. My specific motivation is that I would like to use this pattern for the IR bytecode, and I would be quite unhappy if I had to write out a `Display` impl by hand.

Reply:
Keeping all of impls of the AST nodes separate from the declaration makes it much easier to grok translation functions. You can easily side-by-side see the transformations that much occur. Adding in other data or functionality forces me to try to figure out what information I can discard mentally before doing the comparison.

Reply:
The same sort of principle occurs when programming against the AST itself. I often have two editors open, one of the AST used as a map of the landscape, and the other as the transformation/pass that I am writing. The cleaner the AST, the better the map